sdk/tools/widl/register.c

```diff
@@ -320,7 +359,6 @@ void output_typelib_regscript( const typelib_t *typelib )
     expr = get_attrp( typelib->attrs, ATTR_ID );
     if (expr)
     {
-        sprintf(id_part, "\\%d", expr->cval);
```

Comment:
It appears this part hasn't been correctly synced (and you can see that now, `id_part` is always an empty string, and this could cause problem for the code that is below after the if-block).

Current wine code reads like this:
```c
    if (expr)
    {
        snprintf(id_part, sizeof(id_part), "\\%d", expr->cval);
        resname = strmake("%s\\%d", typelib_name, expr->cval);
    }
```
I don't know whether we now need the `resname` to be initialized (NOTE: the `strmake()` helper should be available from the tools.h file) -- even though it is used at the very end of the function in the `add_output_to_resources.
However, the existing `id_part` initialization is now fixed with the `snprintf`.

Reply:
addressed